training_config/mc/commonsenseqa.jsonnet

```diff
@@ -0,0 +1,52 @@
+local transformer_model = 'roberta-large';
+
+local epochs = 20;
+local batch_size = 64;
+
+local gpu_batch_size = 4;
+local gradient_accumulation_steps = batch_size / gpu_batch_size;
+
+{
+  "dataset_reader": {
+      "type": "commonsenseqa",
+      "transformer_model_name": transformer_model,
+      //"max_instances": 200  // debug setting
+  },
+  "train_data_path": "https://s3.amazonaws.com/commensenseqa/train_rand_split.jsonl",
+  "validation_data_path": "https://s3.amazonaws.com/commensenseqa/dev_rand_split.jsonl",
+  //"test_data_path": "https://s3.amazonaws.com/commensenseqa/test_rand_split_no_answers.jsonl"
+  "model": {
+      "type": "transformer_mc",
+      "transformer_model": transformer_model,
+  },
+  "data_loader": {
+    "sampler": "random",
+    "batch_size": gpu_batch_size
+  },
+  "trainer": {
+    "optimizer": {
+      "type": "huggingface_adamw",
+      "weight_decay": 0.01,
+      "parameter_groups": [[["bias", "LayerNorm\\.weight", "layer_norm\\.weight"], {"weight_decay": 0}]],
```

Comment:
I've seen this in some other configs and I'm just curious what the rational for it is?

Reply:
There are two different spellings for the `LayerNorm.weight` parameters. HF changed from one to the other at some point. This captures both of them.

As for why we turn off weight decay for those parameters? I'm not sure. I think it comes from the original BERT paper where they recommend that. It makes some intuitive sense to me that you wouldn't want to put a decay on biases, but I don't know about layer norm.